src/services/VerifyAccountRequestProcessor.ts

```diff
@@ -343,13 +356,41 @@ export class VerifyAccountRequestProcessor {
 
   calculateExperianCheckResult(verifyResponse: ExperianVerifyResponse, attemptCount?: number): ExperianCheckResult {
   	const personalDetailsScore = verifyResponse?.personalDetailsScore;
-  	if (personalDetailsScore === 9 && !this.isCriticalResponseCode(verifyResponse, this.metrics)) {
-  		return ExperianCheckResults.FULL_MATCH;
-  	} else if (personalDetailsScore !== 9 && attemptCount === undefined) {
+  	if (verifyResponse?.outcome === "STOP" && attemptCount === undefined) {
+  		return undefined;
+  	}
+  	if (personalDetailsScore !== 9 && attemptCount === undefined) {
   		return undefined;
-  	} else {
-  		return ExperianCheckResults.NO_MATCH;
   	}
+  	if (personalDetailsScore === 9 && !this.isCriticalResponseCode(verifyResponse, this.metrics) && verifyResponse?.outcome !== "STOP") {
+  		return ExperianCheckResults.FULL_MATCH;
+  	}
+
+  	return ExperianCheckResults.NO_MATCH;
+  }
+
+  calculateCIs(verifyResponse: ExperianVerifyResponse): string[] | undefined {
+  	const warningsErrors: WarningsErrors[] | undefined = verifyResponse?.warningsErrors;
+  	let cisRequired: string[] = [];
+
+  	const criticalErrors = ["6", "7", "11", "12"];
+  	const warningError  = verifyResponse?.warningsErrors;
+  	if (warningError) {
+  		warningError.forEach(function (value): void {
+  			if (value?.responseType === "error") {
+  				if (criticalErrors.includes(value.responseCode)) {
+  					cisRequired.push("D15");
+  				}
+  			}
+  		}); 
+  	}
+
+  	if (verifyResponse.outcome === "STOP") {
+  		cisRequired.push("D15");
+  	}
+	
+  	cisRequired = [...new Set(cisRequired)];
```

Comment:
Are we ever expecting a CI other than D15?

Reply:
Not currently but there may be an ability to get them in the future